### PR TITLE
Create SubscriberList on EmailAlertApi in background task

### PR DESCRIPTION
**Pull request by dwhenry**
Create a duplicate entry for the subscription on EmailAlertApi.
This is a temporary process as we migrate from GovUkDelivery
to EmailAlertApi. Entries created on EmailAlertApi will be
disabled to avoid duplicate emailing.

https://trello.com/c/Ptoh1EjD/125-1-create-new-topics-in-both-places-with-the-same-govdelivery-topic-id

### CODE REVIEW

File: app/models/email_signup.rb [hunk 1]
```diff
@@ -14,12 +14,18 @@ def initialize(attributes = {})
...
     end
   end
 
   def ensure_govdelivery_topic_exists
-    Whitehall.govuk_delivery_client.topic(feed, description)
+    @ensure_govdelivery_topic_exists ||= Whitehall.govuk_delivery_client.topic(feed, description)
+  end
+
+  def topic_id
+    # should this have some kind of error catching to be on the safe side?
```
Comment (mobaig) [inline app/models/email_signup.rb]: Better this is either removed or reworded into a slightly more assertive TODO.

File: test/unit/email_signup_test.rb [hunk 2]
```diff
@@ -41,4 +53,12 @@ def feed_url(feed_path = "publications.atom")
   def email_signup(feed_path)
     EmailSignup.new(feed: feed_url(feed_path))
   end
+
+  class Response
```
Comment (mobaig) [inline test/unit/email_signup_test.rb]: can we extract this into a module?

File: app/models/email_signup.rb [hunk 3]
```diff
@@ -8,22 +8,24 @@ class EmailSignup
...
-  def local_government
-    ActiveRecord::Type::Boolean.new.type_cast_from_database(@attributes[:local_government])
+  def ensure_govdelivery_topic_exists
+    @ensure_govdelivery_topic_exists ||= Whitehall.govuk_delivery_client.topic(feed, description)
   end
 
-  def ensure_govdelivery_topic_exists
-    Whitehall.govuk_delivery_client.topic(feed, description)
+  def topic_id
+    # should this have some kind of error catching to be on the safe side?
```
Comment (mobaig) [inline app/models/email_signup.rb]: Can we either remove this or reword into a slightly more assertive TODO?

Comment (dwhenry) [inline app/models/email_signup.rb]: removed

File: test/unit/email_signup_test.rb (hunk 2 above)
Comment (dwhenry) [inline test/unit/email_signup_test.rb]: replaced with inline mock

File: app/models/email_signup.rb (hunk 1 above)
Comment (dwhenry) [inline app/models/email_signup.rb]: removed

**Pull request merged**